server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java

```diff
@@ -148,47 +159,74 @@ void cancelTaskAndDescendants(CancellableTask task, String reason, boolean waitF
+        sendBanParentRequests(childConnections, ActionListener.wrap(() -> {}),
+            subNodes -> new BanParentTaskRequest(parentTaskId, subNodes));
+    }
+
+    private void sendBanParentRequests(Map<String, List<DiscoveryNode>> childConnections, ActionListener<Void> listener,
+                                       Function<List<DiscoveryNode>, BanParentTaskRequest> requestGenerator) {
+        if (childConnections.isEmpty()) {
             listener.onResponse(null);
             return;
         }
-        logger.trace("cancelling task {} on child nodes {}", task.getId(), childNodes);
-        GroupedActionListener<Void> groupedListener =
-            new GroupedActionListener<>(ActionListener.map(listener, r -> null), childNodes.size());
-        final BanParentTaskRequest banRequest = BanParentTaskRequest.createSetBanParentTaskRequest(
-            new TaskId(clusterService.localNode().getId(), task.getId()), reason, waitForCompletion);
-        for (DiscoveryNode node : childNodes) {
-            transportService.sendRequest(node, BAN_PARENT_ACTION_NAME, banRequest,
-                new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
-                    @Override
-                    public void handleResponse(TransportResponse.Empty response) {
-                        groupedListener.onResponse(null);
-                    }
-
-                    @Override
-                    public void handleException(TransportException exp) {
-                        assert ExceptionsHelper.unwrapCause(exp) instanceof ElasticsearchSecurityException == false;
-                        logger.warn("Cannot send ban for tasks with the parent [{}] to the node [{}]", banRequest.parentTaskId, node);
-                        groupedListener.onFailure(exp);
-                    }
-                });
+        final int groupSize = childConnections.entrySet().stream()
+            .mapToInt(e -> e.getKey().equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY) ? e.getValue().size() : 1)
+            .sum();
+        final GroupedActionListener<Void> groupedListener =
+            new GroupedActionListener<>(ActionListener.map(listener, r -> null), groupSize);
+        for (Map.Entry<String, List<DiscoveryNode>> entry : childConnections.entrySet()) {
+            final String clusterAlias = entry.getKey();
+            if (clusterAlias.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)) {
+                final BanParentTaskRequest request = requestGenerator.apply(List.of());
+                for (DiscoveryNode node : entry.getValue()) {
+                    sendBanParentRequest(() -> transportService.getConnection(node), request, groupedListener);
+                }
+            } else {
+                final ArrayList<DiscoveryNode> subNodes = new ArrayList<>(entry.getValue());
+                final DiscoveryNode targetNode = subNodes.remove(0);
+                if (targetNode.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
In case of "proxy mode" connections (see also `ProxyConnectionStrategy`), `targetNode.getVersion()` will always return `Version.CURRENT.minimumCompatibilityVersion()` AFAICS, which means that this request won't be send to those nodes.

The actual version of the node on the other side is available using `channel.getVersion()`.
On the other hand `channel.getNode` will return the (possibly fake) `DiscoveryNode` object that was used to create the connection.


/cc: @tbrooks8 This is quite trappy, anything we can change in the transport  in the short term to avoid this?

Reply:
I've modified the [handshaking](https://github.com/elastic/elasticsearch/pull/55779/files#diff-3b8055db0941b6b23392b71384b69678R924) to update the version of the target node with its actual version. I also exposed the [proxy node](https://github.com/elastic/elasticsearch/pull/55779/files#diff-d7e2c78464b3537c4f300288028b8238R159) so that we can check the version of the proxy node. I think we are good here.